File: frontend/src/api/account.ts
import { api } from './client'
import type { User } from './auth'

export const FEEDBACK_CATEGORIES = ['Feature Request', 'App Bug / Error', 'Plant Floor Issue', 'General Feedback'] as const

export interface MyFeedback {
  id: number
  timestamp: string | null
  category: string
  suggestion: string
  status: string
  admin_notes: string | null
}

export const accountApi = {
  updateCredentials: (fullName: string, username: string, pin: string) =>
    api.put<User>('/account/credentials', { full_name: fullName, username, pin }),
  updateTheme: (theme: string) => api.put<User>('/account/theme', { theme }),
  uploadAvatar: (file: File) => {
    const formData = new FormData()
    formData.append('file', file)
    return api.postForm<User>('/account/avatar', formData)
  },
  avatarUrl: (filename: string) => `/api/account/avatar/${encodeURIComponent(filename)}`,

  submitFeedback: (category: string, suggestion: string) =>
    api.post<{ ok: boolean }>('/account/feedback', { category, suggestion }),
  myFeedback: () => api.get<MyFeedback[]>('/account/feedback'),
}

File: api/schemas/account.py
from pydantic import BaseModel


class UpdateCredentialsRequest(BaseModel):
    full_name: str
    username: str
    pin: str = ""  # blank means "leave unchanged" - see crud.update_user_credentials


class UpdateThemeRequest(BaseModel):
    theme: str


class SubmitFeedbackRequest(BaseModel):
    category: str
    suggestion: str


class MyFeedbackOut(BaseModel):
    id: int
    timestamp: str | None
    category: str
    suggestion: str
    status: str
    admin_notes: str | None
